helix-core/src/main/java/org/apache/helix/tools/ClusterVerifiers/BestPossibleExternalViewVerifier.java

```diff
@@ -78,8 +78,8 @@
    */
   @Deprecated
   public BestPossibleExternalViewVerifier(String zkAddr, String clusterName, Set<String> resources,
-      Map<String, Map<String, String>> errStates, Set<String> expectLiveInstances) {
-    super(zkAddr, clusterName);
+      Map<String, Map<String, String>> errStates, Set<String> expectLiveInstances, int waitTillVerify) {
```

Comment:
Could you please add some doc about the rationale of adding waiting time?

Reply:
Good point, let me do it.

Reply:
This is added.

```
   /**
     * The class of verify() methods in this class and its subclass such as 
     * BestPossibleExternalViewVerifier is intend to wait for the cluster converging to a stable 
     * state after changes in the cluster. However, after making changes, it would take some time
     * till controller taking the changes in. Thus, if we verify() too early, before controller
     * taking the changes, the class may mistake the previous stable cluster state as new (expected)
     * stable state. This would cause various issues. Thus, we supply a waitPeriod before starting
     * to validate next expected state to avoid this pre-mature stable state validation.
     */
    public B setWaitTillVerify(int waitPeriod) {
      _waitPeriodTillVerify = waitPeriod;
      return (B) this;
    }

```